src/posthog-surveys.ts

```diff
@@ -67,19 +63,21 @@ export class PostHogSurveys {
         this.loadIfEnabled()
     }
 
-    loadIfEnabled() {
+    loadIfEnabled = () => {
         const surveysGenerator = assignableWindow?.extendPostHogWithSurveys
 
         if (!this.instance.config.disable_surveys && this._decideServerResponse && !surveysGenerator) {
-            if (this._surveyEventReceiver == null) {
-                this._surveyEventReceiver = new SurveyEventReceiver(this.instance.persistence)
-            }
             this.instance.requestRouter.loadScript('/static/surveys.js', (err) => {
                 if (err) {
                     return logger.error(`Could not load surveys script`, err)
                 }
 
                 assignableWindow.extendPostHogWithSurveys(this.instance)
+                if (this._surveyEventReceiver == null) {
+                    const SurveyEventReceiver = assignableWindow?.__PosthogExtensions__?.SurveysEventReceiver
+
+                    this._surveyEventReceiver = new SurveyEventReceiver(this.instance.persistence)
```

Comment:
and then here's we grab the survey event receiver from the lazily loaded extension so we can init it